### ISSUE TITLE
Questions on AUPRC example

### ISSUE THREAD

**Issue by berkuva** [Optimization-AI/LibAUC#14]
What does the parameter `index_s` for APLoss_SH indicate?

Does `posNum` indicate the number of positive samples in each batch, training data, or all data?

**Comment by GangLii** [comment]
In the current released version, `index_s` for APLoss_SH indicates the indices of batched positive samples. The index should be the global index. So we can maintain `u` vector for all the positive samples. If you pass in all the batched samples' indices including positive samples and negative samples, please add this line  `index_s = index_s[y_true == 1].reshape(-1)`. Nowadays, we are refining the lib to make it easier to use, and we will add this part and we will provide some more tutorials about the algorithm and usage.
Usually, `posNum` refers to the number of positive samples in each batch. To make sure, Please specify where the parameter comes from.

**Comment by yzhuoning** [comment]
In version `1.1.8`,  `index_s` in APLoss refers to the indices for positive samples. However, we don't conduct positive indexing inside the training loop but integrate it with the loss function. You can find this operation in the code of `APLoss_SH` class. 

`posNum` refers to the number of positive samples in each **min-batch** (for training data), which is a parameter to control the `sampler` function. For testing set, we don't resampling data.

**Comment by GloryyrolG** [comment]
Hi @yzhuoning et al.,

Thx for ur great work. I noticed in the tutorial notebook u provided for `SOAP` and `APLoss`, it seems the method suffers from severe overfitting, i.e., test AP changes little starting from the beginning and even decreases. May I know if it is a bug? Or I may miss sth.

Looking forward to ur response.


Thx & regards,

![image](https://github.com/Optimization-AI/LibAUC/assets/33202229/2eb29b75-991d-4e85-9d0c-a87cce0b3f6c)

**Comment by optmai** [comment]
This is possible especially when you have a small-scale but highly skewed dataset. To alleviate this issue, model selection (e.g., early stopping) could be used. Thanks!

**Comment by optmai** [comment]
If you have any other questions, please let us know.
